src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -292,9 +292,11 @@ public function configureOptions(OptionsResolver $resolver)
             return;
         };
 
-        $choiceListNormalizer = function (Options $options, $choiceList) use ($choiceListFactory) {
+        // Hack to use the $this variable inside a Closure with PHP 5.3.
+        $that = $this;
+        $choiceListNormalizer = function (Options $options, $choiceList) use ($choiceListFactory, $that) {
             if ($choiceList) {
-                @trigger_error('The "choice_list" option is deprecated since version 2.7 and will be removed in 3.0. Use "choice_loader" instead.', E_USER_DEPRECATED);
+                @trigger_error(sprintf('The "choice_list" option of the "%s" form type (%s) is deprecated since version 2.7 and will be removed in 3.0. Use "choice_loader" instead.', $that->getName(), __CLASS__), E_USER_DEPRECATED);
```

Comment:
since we know that getName is deprecated in 2.8, shouldn't we only tell about `__CLASS__`? At least that's what should be done in 2.8, isn't it?